src/autopas/iterators/ParticleIterator.h

```diff
@@ -57,18 +57,20 @@ class ParticleIterator : public ParticleIteratorInterfaceImpl<Particle, modifiab
    * Can be nullptr if the behavior is haloAndOwned.
    * @param behavior The IteratorBehavior that specifies which type of cells shall be iterated through.
    * @param additionalVectorsToIterate Thread buffers of additional Particle Vector to iterate over.
+   * @param forceSequential Whether to force the iterator to behave as if it is not parallel.
    */
   ParticleIterator(CellVecType *cont, const CellBorderAndFlagManagerType *flagManager, IteratorBehavior behavior,
-                   ParticleVecType *additionalVectorsToIterate)
+                   ParticleVecType *additionalVectorsToIterate, bool forceSequential)
       : _vectorOfCells(cont),
         _iteratorAcrossCells(cont->begin()),
         _iteratorWithinOneCell(cont->begin()->begin()),
         _flagManager(flagManager),
         _behavior(behavior),
+        _forceSequential(forceSequential),
         _additionalParticleVectorToIterateState(additionalVectorsToIterate
                                                     ? AdditionalParticleVectorToIterateState::notStarted
                                                     : AdditionalParticleVectorToIterateState::ignore),
-        _additionalVectorIndex(autopas_get_thread_num()),
+        _additionalVectorIndex(forceSequential ? autopas_get_thread_num() : 0),
```

Comment:
```suggestion
        _additionalVectorIndex(forceSequential ? 0 : autopas_get_thread_num()),
```

Reply:
do you test the case, where you add particles with more threads than with which you iterate (to test the correct iteration using the `_additionalVectorIndex`?)

Reply:
Yes this is tested via three different tests in `ParticleIteratorTest`